sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobUrlParts.java

```diff
@@ -267,6 +273,8 @@ public URL toUrl() {
      * is no path element for the container, the name of this blob in the root container will be set as the
      * containerName field in the resulting {@code BlobURLParts}.</p>
      *
+     * <p>Blob name is encoded to UTF-8 using the {@link com.azure.storage.common.Utility#urlEncode(String)} method.</p>
```

Comment:
Another place that feels likely to give us double encoding since we are taking in a URL

Reply:
See answer about overloads and double encoding above.